### PR TITLE
fixing some basic simplifications change groups

### PR DESCRIPTION
**Pull request by aelnaiem**

### CODE REVIEW

File: lib/simplifyExpression.js [hunk 1]
```diff
@@ -67,13 +68,6 @@ function stepThrough(node, debug=false) {
     }
   }
 
-  // Update the last step, in case we returned something simplified the last
-  // time that didn't count as a step.
-  if (steps.length > 0) {
```
Comment (aelnaiem) [inline lib/simplifyExpression.js]: This change actually breaks this test `subtraction support -------5 -> -5`, but this is a confusing step. Can you explain to me what's going on and why this is necessary?

Comment (evykassirer) [inline lib/simplifyExpression.js]: the idea is nested unary minuses cancel out - that's what it was testing

not sure why it isn't counted as a step though - maybe it happens in flatten?

File: lib/PolynomialTermOperations.js [hunk 2]
```diff
@@ -195,30 +195,29 @@ PolynomialTermOperations.multiplyLikeTerms = function(node) {
...
-      return newTermWithCeoff;
-    }
-    else {
-      return polyNode;
+
+      newNode.args[i].changeGroup = changeGroup;
+      node.args[i].changeGroup = changeGroup; // note that this is the "oldNode"
+
+      change = true;
+      changeGroup++;
```
Comment (evykassirer) [inline lib/PolynomialTermOperations.js]: i like that they're different colours, cool

**Comment by evykassirer** [comment]
mostly looks good, just confused about how one section works

File: lib/EquationOperations.js [hunk 3]
```diff
@@ -164,8 +164,8 @@ EquationOperations.isolateSymbolOnLeftSide = function(equation, symbolName) {
     else if (leftNode.op === '/') {
       // The non symbol term is always a fraction because it's the
       // coefficient of our symbol term.
```
Comment (evykassirer) [inline lib/EquationOperations.js]: what if the node is 2x / 3?

**Comment by evykassirer** [comment]
👍

**Pull request merged**